### PR TITLE
feat(compiler): validate `@HostBinding` name

### PR DESCRIPTION
**Pull request by vicb**
relates to #7500

Blocked on minor release 2.2

### CODE REVIEW

**Comment by francosioquim** [comment]
relates to #7500

Blocked on minor release 2.2
You can view, comment on, or merge this pull request online at:

   https://github.com/angular/angular/pull/12139

-- Commit Summary --

   * feat(compiler): validate `@HostBinding` name

-- File Changes --

     M modules/@angular/compiler/src/directive_resolver.ts (7)
     M modules/@angular/compiler/test/directive_resolver_spec.ts (38)

-- Patch Links --

https://github.com/angular/angular/pull/12139.patch
https://github.com/angular/angular/pull/12139.diff

-- 
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/angular/angular/pull/12139

**Comment by laco0416** [comment]
Question: Is HostBinding inconsistent with Directive.host? Directive.host can take events like `(submit)` at NgForm

**Comment by vicb** [comment]
> Question: Is HostBinding inconsistent with Directive.host? Directive.host can take events like (submit) at NgForm

Read the PR code :)

File: modules/@angular/compiler/src/directive_resolver.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ export class DirectiveResolver {
         } else if (a instanceof HostBinding) {
           const hostBinding: HostBinding = a;
           if (hostBinding.hostPropertyName) {
+            const startWith = hostBinding.hostPropertyName[0];
+            if (startWith === '(') {
+              throw new Error(`@HostBinding can not bind to events. Use @HostListener instead.`);
+            } else if (startWith === '[') {
+              throw new Error(
+                  `@HostBinding parameter should be a property name, 'class.<name>' or 'attr.<name>'.`);
```
Comment (vicb) [inline modules/@angular/compiler/src/directive_resolver.ts]: @StephenFluin should I use an oxford comma here ?

**Comment by laco0416** [comment]
@vicb thank you. I understand.

File: modules/@angular/compiler/src/directive_resolver.ts (hunk 1 above)
Comment (robwormald) [inline modules/@angular/compiler/src/directive_resolver.ts]: yes!

Comment (vicb) [inline modules/@angular/compiler/src/directive_resolver.ts]: It makes feel so smart to use the term "oxford comma" - that Stephen explained me yesterday

**Pull request merged**